server/src/artifact/artifact.service.ts

```diff
@@ -1,78 +1,103 @@
+          throw e;
+        } else {
+          const errorMessage = `An error occured while constructing and looking up folder for artifact. Perhaps the artifact name or id is wrong. ${JSON.stringify(e)}`;
+          console.log(errorMessage);
 
-        throw new HttpException({
-          code: 'SHAREPOINT_FOLDER_ERROR',
-          title: 'Bad Sharepoint folder lookup',
-          detail: errorMessage,
-          meta: {
-            artifactName: artifactName,
-            artifactsId: id,
-          }
-        }, HttpStatus.INTERNAL_SERVER_ERROR);
+          throw new HttpException({
+            code: 'SHAREPOINT_FOLDER_ERROR',
+            title: 'Bad Sharepoint folder lookup',
+            detail: errorMessage,
+            meta: {
+              relativeUrl,
+            }
+          }, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
       }
     }
+
+    console.log(`Warning: Tried to load documents for an Artifact but the "relativeUrl" argument was null. Artifact dcp_name is "${dcp_name}"`);
+
+    return [];
   }
 
   /**
    * Injects associated documents into the given artifact
    *
-   * @param      {Object{ dcp_name, dcp_artifactsid }} CRM Artifact with
-   *              dcp_name and dcp_artifactsid properties
-   * @return     {[Object]} The CRM Artifact with the 'documents' property hydrated,
+   * @param      {Object{ dcp_name, dcp_artifactdocumentlocation }} CRM Artifact with
+   *              dcp_name and dcp_artifactdocumentlocation properties
+   * @return     {Object} The CRM Artifact with the 'documents' property hydrated,
    *              if associated documents exist in CRM.
    */
   public async artifactWithDocuments(projectArtifact: any) {
+    // Unlike the package `dcp_package_SharePointDocumentLocations` property, the `dcp_artifactdocumentlocation`
+    // property is a single-valued property, a string representing the absolute Sharepoint URL.
+    // So we have to split the string to get the "relative URL" portion.
     const {
       dcp_name,
-      dcp_artifactsid,
+      dcp_artifactdocumentlocation,
     } = projectArtifact;
 
-    try {
-      return {
-        ...projectArtifact,
-        documents: await this.findArtifactSharepointDocuments(dcp_name, dcp_artifactsid),
-      };
-    } catch (e) {
-      const errorMessage = `Error loading documents for artifact ${dcp_name}. ${JSON.stringify(e)}`;
-      console.log(errorMessage);
+    if (dcp_artifactdocumentlocation) {
+      try {
+        // Here we assume that the "entity" segment of the dcp_artifactdocumentlocation URL
+        // will be dcp_artifacts, since it is a artifact document location.
+        // Also that the segment divides the absolute URL into two tokens.
+        // Example value for dcp_artifactdocumentlocation:
+        // https://nyco365.sharepoint.com/sites/dcppfsuat2/dcp_artifacts/P2016K0021 - Area Map - 1_77A5253923FBE911A9BC001DD8308EF1
+        const relativeUrl = dcp_artifactdocumentlocation.split('/dcp_artifacts/')[1];
```

Comment:
Is there someting about this assumption that worries you? Seems reasonable to me.

Reply:
What happens when there's not documentlocation? Is it a silent "failure"? Why call the method if the required information isn't there?

Reply:
Yeah, the unfortunate part is it's a silent failure. I don't know what we can do about that though. 

For packages, when that happens I log a warning, since usually by the time a package is public, it will have the document location initiated even if no files are uploaded. And if document location is null, that is an oddity. 

For artifacts, I can't determine the difference between no file uploaded intentionally, which happens often, and some other error causing the artifact document location to be null. So I can't log a warning without spamming  server logs unnecessarily